sources/HUBViewController.m

```diff
@@ -708,19 +706,16 @@ - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cell
     
     HUBComponentCollectionViewCell * const cell = [collectionView dequeueReusableCellWithReuseIdentifier:cellReuseIdentifier
                                                                                             forIndexPath:indexPath];
-    
-    if (cell.component == nil) {
-        HUBComponentWrapper * const componentWrapper = [self.componentReusePool componentWrapperForModel:componentModel
-                                                                                                delegate:self
-                                                                                                  parent:nil];
-        
-        self.componentWrappersByCellIdentifier[cell.identifier] = componentWrapper;
-        cell.component = componentWrapper;
-        [componentWrapper viewDidMoveToSuperview:cell];
-        [self didAddComponentWrapper:componentWrapper];
-    }
-    
-    HUBComponentWrapper * const componentWrapper = [self componentWrapperFromCell:cell];
+
+    HUBComponentWrapper * const componentWrapper = [self.componentReusePool componentWrapperForModel:componentModel
+                                                                                            delegate:self
+                                                                                              parent:nil];
+
+    self.componentWrappersByCellIdentifier[cell.identifier] = componentWrapper;
```

Comment:
Sorry, I don't get it. What is the point of removing object from the dictionary right before setting a new object for the same key?